File: TidbitsTrivia/Core/Models/TriviaCategory.swift
import SwiftUI

/// A trivia category. Categories map Wikipedia content domains onto a
/// fixed, recognizable set with a fixed pop color (no ad-hoc hues).
struct TriviaCategory: Identifiable, Hashable, Sendable {
    let id: String
    let name: String
    let symbol: String   // SF Symbol
    let colorIndex: Int  // index into Tidbits.Palette.pops
    let blurb: String

    var color: Color { Tidbits.Palette.pops[colorIndex % Tidbits.Palette.pops.count] }

    /// The v1 category set. Each maps to Wikipedia category trees /
    /// Vital-Article buckets in the corpus pipeline (see DATA-CONTRACT).
    static let all: [TriviaCategory] = [
        .init(id: "mixed",     name: "Mixed Bag",   symbol: "shuffle",            colorIndex: 0, blurb: "A little of everything."),
        .init(id: "history",   name: "History",     symbol: "scroll.fill",        colorIndex: 1, blurb: "People, places, and the past."),
        .init(id: "science",   name: "Science",     symbol: "atom",               colorIndex: 3, blurb: "How the universe works."),
        .init(id: "geography", name: "Geography",   symbol: "globe.americas.fill",colorIndex: 4, blurb: "The whole wide world."),
        .init(id: "arts",      name: "Arts & Lit",  symbol: "theatermasks.fill",  colorIndex: 5, blurb: "Books, art, and culture."),
        .init(id: "screen",    name: "Film & TV",   symbol: "film.fill",          colorIndex: 0, blurb: "The big and small screen."),
        .init(id: "music",     name: "Music",       symbol: "music.note",         colorIndex: 2, blurb: "From Bach to beats."),
        .init(id: "sports",    name: "Sports",      symbol: "sportscourt.fill",   colorIndex: 1, blurb: "Games and the greats."),
    ]

    static func named(_ id: String) -> TriviaCategory {
        all.first { $0.id == id } ?? all[0]
    }
}

File: windows/Tidbits.Core/Models/TriviaCategory.cs
namespace Tidbits.Core.Models;

/// A trivia category (port of Core/Models/TriviaCategory.swift). `Symbol` +
/// `ColorIndex` are semantic hints; the Avalonia layer maps them to Windows icons
/// and the shared palette. Keep the IDs in lockstep with the other platforms.
public sealed record TriviaCategory(string Id, string Name, string Symbol, int ColorIndex, string Blurb)
{
    public static readonly IReadOnlyList<TriviaCategory> All = new[]
    {
        new TriviaCategory("mixed",     "Mixed Bag",  "shuffle",     0, "A little of everything."),
        new TriviaCategory("history",   "History",    "scroll",      1, "People, places, and the past."),
        new TriviaCategory("science",   "Science",    "atom",        3, "How the universe works."),
        new TriviaCategory("geography", "Geography",  "globe",       4, "The whole wide world."),
        new TriviaCategory("arts",      "Arts & Lit", "theatermasks",5, "Books, art, and culture."),
        new TriviaCategory("screen",    "Film & TV",  "film",        0, "The big and small screen."),
        new TriviaCategory("music",     "Music",      "music",       2, "From Bach to beats."),
        new TriviaCategory("sports",    "Sports",     "sportscourt", 1, "Games and the greats."),
        new TriviaCategory("business",  "Business",   "building2",   3, "Companies and the brands behind them."),
    };

    public static TriviaCategory Named(string id) => All.FirstOrDefault(c => c.Id == id) ?? All[0];
}
